### PR TITLE
feat(pmp): edit repository in settings - backend

### PR DESCRIPTION
**Pull request by ABartoszko**
refs #107

### CODE REVIEW

File: libs/server/repository/api-rest/src/lib/dtos/edit-repository.dto.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class EditRepositoryDto {
+  @ApiProperty()
+  maxLines?: number;
```
Comment (Sikora00) [inline libs/server/repository/api-rest/src/lib/dtos/edit-repository.dto.ts:5]: Add validation as here https://github.com/valueadd-poland/pimp-my-pr/pull/234/files#diff-8b2093dbd0ebd46ebc26f5ecabc4c120

File: libs/server/repository/core/application-services/src/lib/commands/edit-repository/edit-repository.handler.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { RepositoryEntity } from '@pimp-my-pr/server/repository/core/domain';
+
+@CommandHandler(EditRepositoryCommand)
+export class EditRepositoryHandler implements ICommandHandler<EditRepositoryCommand> {
+  constructor(private repositoryRepository: RepositoryRepository) {}
+
+  async execute(command: EditRepositoryCommand): Promise<void> {
+    const { repositoryId, maxLines, maxWaitingTime } = command;
+
+    const repositoryData = await this.repositoryRepository.getById(repositoryId);
```
Comment (Sikora00) [inline libs/server/repository/core/application-services/src/lib/commands/edit-repository/edit-repository.handler.ts:13]: In this case, it is just a repository. In opposite to add action where we wanted to execute construct mothod.

File: libs/server/repository/core/application-services/src/lib/commands/edit-repository/edit-repository.handler.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    const updatedRepository = { ...repositoryData, maxLines, maxWaitingTime } as RepositoryEntity;
```
Comment (Sikora00) [inline libs/server/repository/core/application-services/src/lib/commands/edit-repository/edit-repository.handler.ts:15]: add edit method in the RepositoryEntity end update properties directly like that:
```
edit(writeModel: RepositoryEditWriteModel): void {
  this.maxLines = writeModel.maxLines;
  ...
}
```

File: libs/shared/domain/src/lib/interfaces/repository-edit-write.model.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export interface RepositoryEditWriteModel {
+  maxLines: number;
```
Comment (MaciejSikorski) [inline libs/shared/domain/src/lib/interfaces/repository-edit-write.model.ts:2]: Shouldn't they be optional?

**Pull request merged**